clients/ts/signalr/tests/WebSocketTransport.test.ts

```diff
@@ -0,0 +1,203 @@
+            };
+
+            const message = new TestMessageEvent();
+            message.data = "receive data";
+            TestWebSocket.webSocket.onmessage(message);
+
+            expect(typeof received!).toEqual("string");
+            expect(received!).toEqual("receive data");
+        });
+    });
+
+    it("is closed from WebSocket onclose with error", async () => {
+        await VerifyLogger.run(async (logger) => {
+            (global as any).ErrorEvent = TestEvent;
+            const webSocket = await createAndStartWebSocket(logger);
+
+            let closeCalled: boolean = false;
+            let error: Error;
+            webSocket.onclose = (e) => {
+                closeCalled = true;
+                error = e!;
+            };
+
+            const message = new TestCloseEvent();
+            message.wasClean = false;
+            message.code = 1;
+            message.reason = "just cause";
+            TestWebSocket.webSocket.onclose(message);
+
+            expect(closeCalled).toEqual(true);
+            expect(error!).toEqual(new Error("Websocket closed with status code: 1 (just cause)"));
+
+            await expect(webSocket.send(""))
+                .rejects
+                .toThrow("WebSocket is not in the OPEN state");
+        });
+    });
+
+    it("is closed from WebSocket onclose", async () => {
+        await VerifyLogger.run(async (logger) => {
+            (global as any).ErrorEvent = TestEvent;
+            const webSocket = await createAndStartWebSocket(logger);
+
+            let closeCalled: boolean = false;
+            let error: Error;
+            webSocket.onclose = (e) => {
+                closeCalled = true;
+                error = e!;
+            };
+
+            const message = new TestCloseEvent();
+            message.wasClean = true;
+            message.code = 1000;
+            message.reason = "success";
+            TestWebSocket.webSocket.onclose(message);
+
+            expect(closeCalled).toEqual(true);
+            expect(error!).toEqual(undefined);
+
+            await expect(webSocket.send(""))
+                .rejects
+                .toThrow("WebSocket is not in the OPEN state");
+        });
+    });
+
+    it("is closed from Transport stop", async () => {
+        await VerifyLogger.run(async (logger) => {
+            (global as any).ErrorEvent = TestEvent;
+            const webSocket = await createAndStartWebSocket(logger);
+
+            let closeCalled: boolean = false;
+            let error: Error;
+            webSocket.onclose = (e) => {
+                closeCalled = true;
+                error = e!;
+            };
+
+            await webSocket.stop();
+
+            expect(closeCalled).toEqual(true);
+            expect(error!).toEqual(undefined);
+
+            await expect(webSocket.send(""))
+                .rejects
+                .toThrow("WebSocket is not in the OPEN state");
+        });
+    });
+
+    it("can send data", async () => {
```

Comment:
add one for binary data

Reply:
really just means sending a `Uint8Array` instead of a string and watching it go through